library_generation/generate_composed_library.py

```diff
@@ -0,0 +1,139 @@
+"""
+This script allows generation of libraries that are composed of more than one
+service version. It is achieved by calling `generate_library.sh` without
+postprocessing for all service versions and then calling
+postprocess_library.sh at the end, once all libraries are ready.
+
+Arguments
+--generation_queries: a single string of comma-separated key-value groups separated by a
+pipe | (i.e. the groups are spearated by pipe, while a group's key-values are
+separated by comma). They key-value groups are in the form of `key=value` and will
+be converted to an argument to generate_library.sh (`--key value`).
+  example: "proto_path=google/cloud/asset/v1,destination_path=google-cloud-asset-v1-java,(...)|proto_path=google/cloud/asset/v1p2beta5,destination_path=google-cloud-asset-v1-java,(...)"
+  In this case, generate_library.sh will be called once with
+    generate_library.sh --proto_path google/cloud/asset/v1 --destination_path google-cloud-asset-v1-java
+    and once with
+    generate_library.sh --proto_path google/cloud/asset/v1p2bet5 --destination_path google-cloud-asset-v1p2beta5-java
+--versions_file: a file of newline-separated version strings in the form "module:released-version:current-version". The versions will be applied to the pom.xml files and readmes
+
+Prerequisites
```

Comment:
Does the caller need to run "git clone" for the repository this script creates a commit in?

Reply:
It is optional. If the user specifies `--repository_path` in `main.py`, then the generated source code will be sent to such path. If not specified, the `composed_library` script will download the repository into `output_folder` and send the generated sources there

Reply:
I also added a small explanation

https://github.com/googleapis/sdk-platform-java/blob/1bf473aa5425dc3f0222dca7123cad69f97553a6/library_generation/generate_composed_library.py#L37-L38